### DISCUSSION TITLE
Kuberentes Admin user - root or ansible ?

### DISCUSSION THREAD

**Discussion by MagnaFrisia90** [HCL-TECH-SOFTWARE/connections-automation#261] [Q&A] [answered]
Hello,
I have used the following command to setup kubernetes:
ansible-playbook -i environments/examples/cnx7/quick_start/inventory.ini playbooks/setup-component-pack-infra-only.yml

It seems like this ansible creates a /home/ansible/.kube/config file.
Thus only "ansible" can run "kubectl" on the master node but not root (which I exepected to be the "kubernetes admin")

Is this intended ? 

Regards,
Jan

**Comment by umeli** [comment]
Hi 
in order to run kbuectl as root, you just need to create the .kube folder in your root's home. the just run :
 cp /etc/kubernetes/admin.conf ~/.kube/config
 after that root should be able to run any kubectl commands

**Comment by sabrina-yee** [comment, marked as answer]
yes it's intended.  If you'd like other users to use kubectl, do what @umeli suggested which is also automated via `playbooks/third_party/kubernetes/setup-kubectl.yml` when  `kubectl_user` is added to the inventory all.yml for the target user.

https://github.com/HCL-TECH-SOFTWARE/connections-automation/blob/main/roles/third_party/kubernetes/setup-kubectl/tasks/setup_kubectl.yml
